index.user.js

```diff
@@ -230,6 +230,31 @@ const EnemyCenter = function EnemyCenter(enemy) {
         enemy.m_Sprite.y + enemy.m_Sprite.height / 2
     ];
 }
+const EnemyWillAffectedByBoulder = function EnemyWillAffectedByBoulder(enemy) {
+    if(GAME.m_State.m_AttackManager.m_mapBoulders.size > 0) {
+        let boulder = GAME.m_State.m_AttackManager.m_mapBoulders.values().next().value;
+        let x = EnemyCenter(enemy)[0];
+        let y = EnemyCenter(enemy)[1];
+        if(x > boulder.x && y > boulder.y - (boulder.height / 2) && y < boulder.y + (boulder.height / 2)) {
+            return true;
+        }
+    }
+    return false;
+}
+const DistBetweenPoints = function DistBetweenPoints(x1, y1, x2, y2) {
+    return Math.sqrt( Math.pow((x1-x2), 2) + Math.pow((y1-y2), 2) );
+}
+const AllEnemiesHPNearPoint = function AllEnemiesHPNearPoint(x,  y, radius) {
+    let hp = 0;
+    EnemyManager().m_rgEnemies.forEach((enemy) => {
```

Comment:
I am not an expert on JS, but do you really want to execute a `forEach` here? Couldn't it be async?

Reply:
yes you do since m_rgEnemies is a Map object

Reply:
@meepen so it is guaranteed to not to be async? What about the following approach?
```js
for(var [_, enemy] of EnemyManager().m_rgEnemies) {
        // If something
    hp += enemy.m_nHealth;
}
```